src/libs/Navigation/linkingConfig/getMatchingCentralPaneRouteForState.ts

```diff
@@ -31,8 +31,45 @@ const getTopMostReportIDFromRHP = (state: State): string => {
     return '';
 };
 
+// Function that checks if the given route has a policyID equal to the id provided in the function params
+function hasRouteMatchingPolicyID(route: NavigationPartialRoute<CentralPaneName>, policyID?: string) {
+    return (
+        route?.params &&
+        'params' in route?.params &&
+        route.params.params &&
+        'policyID' in (route.params.params as Record<string, string | undefined>) &&
```

Comment:
Please don't type assert this twice, extract a local variable of desired target type.